### ISSUE TITLE
RNGooglePlaces - Semantic Issue

### ISSUE THREAD

**Issue by joohimks** [tolu360/react-native-google-places#227]
I have been struggling for the past few days with this error:

<img width="266" alt="Screen Shot 2019-05-22 at 12 13 33 PM" src="https://user-images.githubusercontent.com/44742670/58158279-2ab03480-7c8b-11e9-942b-22396a54cdda.png">

The map  is working perfectly fine and redux is managing my locations data states perfectly but the places is not working. I got this error the moment I updated react-native-google-places to 3.0.2. Prior to the update, I used to get this error:
<img width="609" alt="Screen Shot 2019-05-22 at 5 12 36 AM" src="https://user-images.githubusercontent.com/44742670/58158608-ce99e000-7c8b-11e9-8690-45ef3fc8ba50.png">

My API at the console is enabled and billing is on as it should be:
<img width="474" alt="Screen Shot 2019-05-22 at 5 15 10 AM" src="https://user-images.githubusercontent.com/44742670/58158635-deb1bf80-7c8b-11e9-92c4-341110ee5b55.png">

Here is my package.json:
<img width="435" alt="Screen Shot 2019-05-22 at 12 15 38 PM" src="https://user-images.githubusercontent.com/44742670/58158358-56331f00-7c8b-11e9-95cd-83ace379b8fe.png">

Here is my pod file:
<img width="457" alt="Screen Shot 2019-05-22 at 12 16 39 PM" src="https://user-images.githubusercontent.com/44742670/58158423-782ca180-7c8b-11e9-8bc3-3defa9e4e96e.png">

And Here is my pod install output:
<img width="660" alt="Screen Shot 2019-05-22 at 12 09 45 PM" src="https://user-images.githubusercontent.com/44742670/58158375-621ee100-7c8b-11e9-9063-a007e0b70634.png">

I know I am missing something but I hope someone can guide me through the right direction.

Many Thanks

**Comment by joohimks** [comment]
I have updated cocoapods to 1.7.0.rc.2 and the semantic issue disappeared. A new error came up:

'Google-Maps-iOS-Utils/GMUKMLParser.h' file not found

<img width="264" alt="Screen Shot 2019-05-22 at 1 27 02 PM" src="https://user-images.githubusercontent.com/44742670/58163593-546e5900-7c95-11e9-8912-ba4880b56867.png">

**Comment by igorlamos** [comment]
I experienced the same Issue when updating to the latest version.

There is an instruction in the `README` - [here](https://github.com/tolu360/react-native-google-places#install-cocoapods-dependencies):

> Add pod 'GooglePlaces', pod 'GooglePlacePicker' and pod 'GoogleMaps' to your Podfile. **Ensure you pull in version 3.1.0 or higher for both the GooglePlaces and GoogleMaps libraries.**

I have this content in the `Podfile`:
```
pod 'GooglePlacePicker'
pod 'GooglePlaces'
pod 'GoogleMaps'
pod 'Google-Maps-iOS-Utils'
```

the `pod install` command installs this versions:
```
Installing Google-Maps-iOS-Utils (2.1.0)
Installing GoogleMaps (2.5.0)
Installing GooglePlacePicker (2.5.0)
Installing GooglePlaces (2.5.0)
```

I'm also using `react-native-maps` ... the latest version on npm is **0.24.2**, which has 'GoogleMaps' 2.5.0 as a dependency in `podspec`. The code in `master` branch is newer, so I installed the version from `master` branch, where the `GoogleMaps` dependency is `3.0.3`. Then, I have updated this dependency to `3.1.0` (to avoid resolution conflict when installing Pods). I'm using `patch-package` to track these changes inside `node_modules` packages ...

Then I specified required versions in my `Podfile`:
```
pod 'GooglePlacePicker', '~> 3.1.0'
pod 'GooglePlaces', '~> 3.1.0'
pod 'GoogleMaps', '~> 3.1.0'
pod 'Google-Maps-iOS-Utils'
```

and ran `pod install` in the iOS directory, which results in the following versions
```
Installing Google-Maps-iOS-Utils (2.1.0)
Installing GoogleMaps (3.1.0)
Installing GooglePlacePicker (3.1.0)
Installing GooglePlaces (3.1.0)
```

Now everything works. I assume not all steps are required ... but:

**Just make sure `pod install` installs the required package(s) version(s).**

Hope this helps.

**Comment by joohimks** [comment]
@igorlamos I have specified the versions in my podfile as below:
` pod 'GoogleMaps', '~> 3.1.0'
  pod 'GooglePlaces', '~> 3.1.0'
  pod 'GooglePlacePicker', '~> 3.1.0'
  pod 'Google-Maps-iOS-Utils' `

And ran pod update && pod install and here is the output for the update:
`[!] CocoaPods could not find compatible versions for pod "GoogleMaps":
  In Podfile:
    GoogleMaps (~> 3.1.0)

    react-native-google-maps (from `../node_modules/react-native-maps`) was resolved to 0.23.0, which depends on
      GoogleMaps (= 2.5.0)

Specs satisfying the `GoogleMaps (~> 3.1.0), GoogleMaps (= 2.5.0)` dependency were found, but they required a higher minimum deployment target.`

and here is the output for the install "nothing has changed":
`Using GoogleMaps (2.5.0)
Using GooglePlacePicker (2.5.0)
Using GooglePlaces (2.5.0)
Using react-native-google-maps (0.23.0)
Using react-native-maps (0.23.0)`

**Comment by igorlamos** [comment]
If you go through my previous comment step-by-step, you will solve your problem. It's the same as I have.

> I'm also using react-native-maps ... the latest version on npm is 0.24.2, which has 'GoogleMaps' 2.5.0 as a dependency in podspec. **The code in master branch is newer, so I installed the version from master branch, where the GoogleMaps dependency is 3.0.3. Then, I have updated this dependency to 3.1.0 (to avoid resolution conflict when installing Pods).**

**Comment by joohimks** [comment]
I have followed everything as mentioned, Updated podspec file, add , '~> 3.1.0' to my pods and still have the same error.

I believe I am missing the batch part, could you please clarify that part.

Your support is extremely appreciated on this.

**Comment by igorlamos** [comment]
What version of `react-native-maps` are you using right now? Take a look to the file `node_modules/react-native-maps/react-native-google-maps.podspec` ... what do you see on the line: 
```
s.dependency 'GoogleMaps', '3.1.0' <<< What is here?
```

That's the place causing a CocoaPods version resolution problem ... 

1) install `react-native-maps` from master branch
2) update the `node_modules/react-native-maps/react-native-google-maps.podspec` file on L20:
```
s.dependency 'GoogleMaps', '3.1.0'
```
3) remove your `Pods` directory, `Podfile.lock` & run `pod install --repo-update`
4) I hope you will have fun ;)

**Comment by joohimks** [comment]
I am using this version at the podspec:
  s.dependency 'GoogleMaps', '2.5.0'

and I updated it to
s.dependency 'GoogleMaps', '3.1.0'
<img width="408" alt="Screen Shot 2019-05-22 at 3 00 01 PM" src="https://user-images.githubusercontent.com/44742670/58169656-525ec700-7ca2-11e9-95c0-76c460b9a699.png">

<img width="425" alt="Screen Shot 2019-05-22 at 2 59 36 PM" src="https://user-images.githubusercontent.com/44742670/58169676-61457980-7ca2-11e9-9da8-3eb6765fd956.png">

<img width="729" alt="Screen Shot 2019-05-22 at 3 01 14 PM" src="https://user-images.githubusercontent.com/44742670/58169735-8b973700-7ca2-11e9-940b-858d560a9b7d.png">


did everything by the letter and still having the issue.

**Comment by igorlamos** [comment]
Post me your current `package.json` file - especially the line `react-native-maps` ...

**Comment by joohimks** [comment]
Here we go:

<img width="426" alt="Screen Shot 2019-05-22 at 3 06 59 PM" src="https://user-images.githubusercontent.com/44742670/58169981-43c4df80-7ca3-11e9-8fd4-20e594f28713.png">

**Comment by igorlamos** [comment]
I know ... You have to read carefully ;)

You cannot have `react-native-maps` version **0.24.2** in `package.json` and update the `podspec` from 'GoogleMaps' `2.5.0` > `3.1.0`. As you may know, this versions represents [Semantic Versioning](https://semver.org/) ... and updating from `2.5.0` > `3.1.0` indicates breaking change. That's why I wrote about **installing `react-native-maps`** from `master` branch. Because, when you take a look to the [`podspec`](https://github.com/react-native-community/react-native-maps/blob/master/react-native-google-maps.podspec) file in the `master` branch, you will see, that the 'GoogleMaps' dependency version there is set to `3.0.3` - which seems fine to update from `3.0.3` > `3.1.0`. I know that installing a package from `master` branch is not "the best way", but for know (before next `react-native-maps` release) - it's only possible way - I think.

So, the last change you need to do - change the line in `package.json` from:
```json
"react-native-maps": "^0.24.2"
```

to:

```json
"react-native-maps": "https://github.com/react-native-community/react-native-maps#dbf746d66ca1b42f2beb790bfbf4c0e3a74f3279"
```

Then change the dependency version in `podspec` from 3.0.3 > 3.1.0 & Rebuild.

**Comment by joohimks** [comment]
I did the following:
1- updated the `podspec` that is inside `node_modules..`  to 3.1.0. When I ran pod install I get the following error:
`[!] CocoaPods could not find compatible versions for pod "GoogleMaps":
  In Podfile:
    GoogleMaps (~> 3.1.0)

    react-native-google-maps (from `../node_modules/react-native-maps`) was resolved to 0.24.0, which depends on
      GoogleMaps (= 3.0.3)

Specs satisfying the `GoogleMaps (~> 3.1.0), GoogleMaps (= 3.0.3)` dependency were found, but they required a higher minimum deployment target.`

The podspec in localPodspecs keeps changing after any pod command to 3.0.3. 

I don't know how to express my gratitude to your patience and support.

**Comment by igorlamos** [comment]
According me, you're doing something wrong. But for now - I don't know what. Our problem is exactly the same ... and on my side, everything works correctly. When you update the version to 3.1.0 in `react-native-google-maps.podspec`, why Cocoapods displays the 3.0.3 version - where it's coming from? Go over the points once again and carefully .. I think you are missing someting out.

**Comment by joohimks** [comment]
@igorlamos I spent the last 9 hours trying to track the file and until now I have no clue from where it keeps changing to 3.0.3. I will let you know soon :) ... hopefully

**Comment by joohimks** [comment]
It worked finally. Thank you very much @igorlamos

**Comment by brownieboy** [comment]
I had the "CocoaPods could not find compatible versions for pod "GoogleMaps" error on one Mac, but on another it worked just fine.

I just needed to run `pod repo update` to fix it.

**Comment by bericp1** [comment]
Dropping a note here that [`react-native-maps`](https://www.npmjs.com/package/react-native-maps) released version `0.25.0` today which includes their updated dependency on `GoogleMaps` `3.1.0`. So you should be able to resolve this by running:

```
yarn add react-native-maps@^0.25.0
cd ios && pod update && cd ../
```

**Comment by blueneel05** [comment]
I had this error 
`[!] CocoaPods could not find compatible versions for pod "GoogleMaps":
  In Podfile:
    react-native-google-maps (from ../node_modules/react-native-maps) was resolved to 0.24.2, which depends on
      Google-Maps-iOS-Utils (= 2.1.0) was resolved to 2.1.0, which depends on
        GoogleMaps

    react-native-google-maps (from ../node_modules/react-native-maps) was resolved to 0.24.2, which depends on
      GoogleMaps (= 2.5.0)

    react-native-google-places (from ../node_modules/react-native-google-places) was resolved to 3.1.0, which depends on
      GoogleMaps (~> 3.1.0)`
Using below
`yarn add react-native-maps@^0.25.0
cd ios && pod update && cd ../`
Solved the issue. Thank you @bericp1

**Comment by Moimou** [comment]
> I experienced the same Issue when updating to the latest version.
> 
> There is an instruction in the `README` - [here](https://github.com/tolu360/react-native-google-places#install-cocoapods-dependencies):
> 
> > Add pod 'GooglePlaces', pod 'GooglePlacePicker' and pod 'GoogleMaps' to your Podfile. **Ensure you pull in version 3.1.0 or higher for both the GooglePlaces and GoogleMaps libraries.**
> 
> I have this content in the `Podfile`:
> 
> ```
> pod 'GooglePlacePicker'
> pod 'GooglePlaces'
> pod 'GoogleMaps'
> pod 'Google-Maps-iOS-Utils'
> ```
> 
> the `pod install` command installs this versions:
> 
> ```
> Installing Google-Maps-iOS-Utils (2.1.0)
> Installing GoogleMaps (2.5.0)
> Installing GooglePlacePicker (2.5.0)
> Installing GooglePlaces (2.5.0)
> ```
> 
> I'm also using `react-native-maps` ... the latest version on npm is **0.24.2**, which has 'GoogleMaps' 2.5.0 as a dependency in `podspec`. The code in `master` branch is newer, so I installed the version from `master` branch, where the `GoogleMaps` dependency is `3.0.3`. Then, I have updated this dependency to `3.1.0` (to avoid resolution conflict when installing Pods). I'm using `patch-package` to track these changes inside `node_modules` packages ...
> 
> Then I specified required versions in my `Podfile`:
> 
> ```
> pod 'GooglePlacePicker', '~> 3.1.0'
> pod 'GooglePlaces', '~> 3.1.0'
> pod 'GoogleMaps', '~> 3.1.0'
> pod 'Google-Maps-iOS-Utils'
> ```
> 
> and ran `pod install` in the iOS directory, which results in the following versions
> 
> ```
> Installing Google-Maps-iOS-Utils (2.1.0)
> Installing GoogleMaps (3.1.0)
> Installing GooglePlacePicker (3.1.0)
> Installing GooglePlaces (3.1.0)
> ```
> 
> Now everything works. I assume not all steps are required ... but:
> 
> **Just make sure `pod install` installs the required package(s) version(s).**
> 
> Hope this helps.
